falcon/api.py

```diff
@@ -28,6 +28,21 @@
 from falcon.util.misc import get_argnames
 
 
+def _make_router_search(router):
+    """Given a router instance returns a function that accepts a request
+    and invokes the router's find method.
+    """
+    arg_names = get_argnames(router.find)
+    supports_req = 'req' in arg_names and len(arg_names) > 1
+
+    def search(request, pass_req=supports_req):
+        if pass_req:
+            return router.find(request.path, req=request)
+        return router.find(request.path)
+
+    return search
```

Comment:
In the case that the router supports the new interface, you can just `return router.find` directly; no need to shim it.